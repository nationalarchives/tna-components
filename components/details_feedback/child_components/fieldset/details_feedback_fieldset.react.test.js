import React from 'react';
import TestRenderer from 'react-test-renderer';
import renderer from 'react-test-renderer';
import Fieldset from './details_feedback_fieldset.react';

let testRenderer, testRendererToJson;

// Snapshot test 1
it('renders the checkbox component correctly', () => {
  const items = [
    'Legend heading',
    'Lorem Ipsum is simply dummy text of the printing and typesetting industry.'
  ];
  const tree = renderer
    .create(
      <Fieldset legendText={items[0]}>
        <p>{items[1]}</p>
      </Fieldset>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe("Check DOM element and it's attributes", () => {
  const items = [
    'Legend heading',
    'Lorem Ipsum is simply dummy text of the printing and typesetting industry.'
  ];
  testRenderer = TestRenderer.create(
    <Fieldset legendText={items[0]}>
      <p>{items[1]}</p>
    </Fieldset>
  );

  testRendererToJson = testRenderer.toJSON();
  it('Should have an <fieldset> tag', () => {
    expect(testRendererToJson.type).toBe('fieldset');
    expect(testRendererToJson.hasOwnProperty('children')).toBeTruthy();
    expect(testRendererToJson.children.length).toBe(2);
  });
  it('Should have props', () => {
    expect(testRendererToJson.props.hasOwnProperty('className')).toBeTruthy();
    expect(testRendererToJson.props.hasOwnProperty('id')).toBeTruthy();
    expect(testRendererToJson.props.hasOwnProperty('className')).toBeTruthy();
  });
  it('Should have legend tag', () => {
    expect(testRendererToJson.children[0].type).toBe('legend');
    expect(
      testRendererToJson.children[0].hasOwnProperty('children')
    ).toBeTruthy();
  });
});
