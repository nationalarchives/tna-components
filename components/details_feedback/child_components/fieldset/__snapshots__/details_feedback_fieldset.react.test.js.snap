// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the checkbox component correctly 1`] = `
<fieldset
  className={null}
  id={null}
>
  <legend>
    Legend heading
  </legend>
  <p>
    Lorem Ipsum is simply dummy text of the printing and typesetting industry.
  </p>
</fieldset>
`;
